### PR TITLE
INTMDB-122: Private Endpoints

### PR DESCRIPTION
**Pull request by coderGo93**
## Description

Added two files for private endpoints and its tests and the old ones changed the name to `_deprecated` at the end of file.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by themantissa — APPROVED** [review]
LGTM, thank you!  Will add a reviewer on the MDB side since the rename will be a breaking change.  Also since we now have Jira can you start putting the internal ticket in the PR subject, e.g. this one is INTMDB-122.  Will help with tracking.

**Comment by themantissa** [comment]
@gssbzn since this is a breaking change and Angel is OOO would you mind reviewing this (or assigning someone else)?

@shum  FYI for you since you are close to this one but feel free to input if you have any.

File: mongodbatlas/private_endpoints.go [hunk 1]
```diff
@@ -29,25 +29,31 @@ var _ PrivateEndpointsService = &PrivateEndpointsServiceOp{}
...
+	PrivateEndpointResourceID     string `json:"privateEndpointResourceId,omitempty"`     // Unique identifier of the private endpoint.
+	DeleteRequested               *bool  `json:"deleteRequested,omitempty"`               // Indicates if Atlas received a request to remove the interface endpoint from the private endpoint connection.
+	ErrorMessage                  string `json:"errorMessage,omitempty"`                  // Error message pertaining to the interface endpoint. Returns null if there are no errors.
+	ConnectionStatus              string `json:"connectionStatus,omitempty"`              // Status of the interface endpoint: NONE, PENDING_ACCEPTANCE, PENDING, AVAILABLE, REJECTED, DELETING.
 }
 
-// Create one private endpoint connection in an Atlas project.
-// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoint-create-one-private-endpoint-connection/
+// Create one private endpoint service for AWS or Azure in an Atlas project.
+// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-create-one/
```
Comment (gssbzn) [inline mongodbatlas/private_endpoints.go:56]: [nit] to improve how this renders it should be, same with the others but not a blocker
```suggestion
// Create one private endpoint service for AWS or Azure in an Atlas project.
//
// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-create-one/
```

File: mongodbatlas/private_endpoints.go [hunk 2]
```diff
@@ -128,18 +142,21 @@ func (s *PrivateEndpointsServiceOp) List(ctx context.Context, groupID string, li
 	return *root, resp, nil
 }
 
-// Delete removes one private endpoint connection in an Atlas project.
-// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoint-delete-one-private-endpoint-connection/
-func (s *PrivateEndpointsServiceOp) Delete(ctx context.Context, groupID, privateLinkID string) (*Response, error) {
+// Delete one private endpoint service for AWS or Azure in an Atlas project.
+// See morehttps://docs.atlas.mongodb.com/reference/api/private-endpoints-service-delete-one/
```
Comment (gssbzn) [inline mongodbatlas/private_endpoints.go:146]: [nit] same as before but this one on the typo `morehttps`
```suggestion
// Delete one private endpoint service for AWS or Azure in an Atlas project.
//
// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints-service-delete-one/
```

File: mongodbatlas/private_endpoints.go [hunk 3]
```diff
@@ -179,21 +199,24 @@ func (s *PrivateEndpointsServiceOp) AddOneInterfaceEndpoint(ctx context.Context,
 	return root, resp, err
 }
 
-// GetOneInterfaceEndpoint retrieves one interface endpoint in a private endpoint connection in an Atlas project.
-// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoint-get-one-interface-endpoint/
-func (s *PrivateEndpointsServiceOp) GetOneInterfaceEndpoint(ctx context.Context, groupID, privateLinkID, interfaceEndpointID string) (*InterfaceEndpointConnection, *Response, error) {
+// GetOnePrivateEndpoint retrieve details for one private endpoint for AWS or Azure in an Atlas project..
+// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints-endpoint-get-one/
```
Comment (gssbzn) [inline mongodbatlas/private_endpoints.go:203]: [nit] double dots
```suggestion
// GetOnePrivateEndpoint retrieve details for one private endpoint for AWS or Azure in an Atlas project.
//
// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints-endpoint-get-one/
```

**Review by gssbzn — DISMISSED** [review]
there are a couple of typos on the docs but not a blocker from me

**Comment by themantissa** [comment]
@coderGo93 if you can make a few of the minor changes suggested I think this is done!

**Review by themantissa — APPROVED** [review]
LGTM, thanks for fixing.

**Comment by themantissa** [comment]
@gssbzn the typos are fixed the update lost your approval.  Can you submit again?

**Review by gssbzn — APPROVED** [review]
LGTM thanks for the fixes

**Pull request merged**